### PR TITLE
chore: Adding tooltip for Select components 

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Enhancement

This PR resolves [CX-2231]

### Description

This PR is the continuation of [closed PR ](https://github.com/artsy/eigen/pull/5928)

image

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-Tooltip for select components on storybooks - sam

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-2231]: https://artsyproduct.atlassian.net/browse/CX-2231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Tooltip for select components on storybooks - sam)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6c546422b6207dcbd0bf8677a58591361e455e94
</p>

**Review by olerichter00 — APPROVED** [review]
👍

File: src/palette/elements/Select/Select.tsx [hunk 1]
```diff
@@ -26,10 +26,12 @@ export interface SelectProps<ValueType> {
   enableSearch?: boolean
   maxModalHeight?: number
   hasError?: boolean
+  tooltipText?: string
   testID?: string
   onSelectValue(value: ValueType, index: number): void
   renderButton?(args: { selectedValue: ValueType | null; onPress(): void }): JSX.Element
   renderItemLabel?(value: SelectOption<ValueType>): JSX.Element
+  onTooltipPress?(): void
```
Comment (pvinis) [inline src/palette/elements/Select/Select.tsx:34]: this name feels much nicer.

File: src/palette/elements/Select/Select.tsx [hunk 2]
```diff
@@ -26,10 +26,12 @@ export interface SelectProps<ValueType> {
   enableSearch?: boolean
   maxModalHeight?: number
   hasError?: boolean
+  tooltipText?: string
```
Comment (pvinis) [inline src/palette/elements/Select/Select.tsx:29]: tooltip still feels wrong, because this is not a tooltip.

Comment (rajsam003) [reply src/palette/elements/Select/Select.tsx:29]: how about `info`?

**Pull request merged**